packages/core/src/shared/telemetry/vscodeTelemetry.json

```diff
@@ -1243,6 +1249,60 @@
                     "required": true
                 }
             ]
+        },
+        {
+            "name": "threatComposer_created",
+            "description": "Called after a new threat composer file is created using command pallet or New File option",
+            "metadata": [
+                {
+                    "type": "id",
+                    "required": true
+                }
+            ]
+        },
+        {
+            "name": "threatComposer_opened",
+            "description": "Called after opening a threat composer file",
+            "metadata": [
+                {
+                    "type": "id",
+                    "required": true
+                }
+            ]
+        },
+        {
+            "name": "threatComposer_fileSaved",
+            "description": "Called after a threat composer file has been saved",
+            "metadata": [
+                {
+                    "type": "id",
+                    "required": true
+                },
+                {
+                    "type": "saveType",
+                    "required": true
+                }
+            ]
+        },
+        {
+            "name": "threatComposer_closed",
+            "description": "Called after closing a threat composer file",
+            "metadata": [
+                {
+                    "type": "id",
+                    "required": true
+                }
+            ]
+        },
+        {
+            "name": "threatComposer_error",
```

Comment:
every metric can store a `result` and `reason` field, and this is strongly recommended, see https://github.com/aws/aws-toolkit-vscode/blob/master/docs/telemetry.md

 `telemetry.run()` helps with this, and also results in improved UX when failures occur.